src/Service/S3/tests/Integration/S3ClientTest.php

```diff
@@ -197,6 +199,30 @@ public function testCreateDirectory()
         self::assertEquals(200, $info['status']);
     }
 
+    public function testCustomEndpoint()
+    {
+        $callback = function ($method, $url, $options) {
+            $this->assertEquals('PUT', $method);
+            $this->assertEquals('https://fra1.digitaloceanspaces.com/foo/bar/', $url);
+
+            return new SimpleMockedResponse();
+        };
+
+        $s3 = new S3Client([
+            'endpoint' => 'https://fra1.digitaloceanspaces.com',
```

Comment:
Test will failed only after that

Reply:
Yes. Then it will fail with:

```
AsyncAws\Core\Exception\UnsupportedRegion : The region "fra1" is not supported by "S3".
 /Users/tobias/Workspace/PHPStorm/aws/src/Service/S3/src/S3Client.php:805
 /Users/tobias/Workspace/PHPStorm/aws/src/Core/src/AbstractApi.php:196
 /Users/tobias/Workspace/PHPStorm/aws/src/Service/S3/src/S3Client.php:661
 /Users/tobias/Workspace/PHPStorm/aws/src/Core/src/AbstractApi.php:113
 /Users/tobias/Workspace/PHPStorm/aws/src/Service/S3/src/S3Client.php:584
 /Users/tobias/Workspace/PHPStorm/aws/src/Service/S3/tests/Integration/S3ClientTest.php:218
 
```